octavia-cli/octavia_cli/apply/resources.py

```diff
@@ -587,7 +588,7 @@ def _check_for_legacy_connection_configuration_keys(self, configuration_to_check
         self._check_for_invalid_configuration_keys(
             configuration_to_check, {"syncCatalog", "namespaceDefinition", "namespaceFormat", "resourceRequirements"}, error_message
         )
-        self._check_for_invalid_configuration_keys(configuration_to_check["schedule"], {"timeUnit"}, error_message)
+        self._check_for_invalid_configuration_keys(configuration_to_check.get("schedule", {}), {"timeUnit"}, error_message)
```

Comment:
Why don't move this function to the if block?

Reply:
There's no `if` block in this function 😄

Reply:
forget it